easybuild/easyblocks/x/xalt.py

```diff
@@ -0,0 +1,160 @@
+##
+# Copyright 2020 NVIDIA
+#
+# This file is triple-licensed under GPLv2 (see below), MIT, and
+# BSD three-clause licenses.
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for XALT, implemented as an easyblock
+@author: Scott McMillan (NVIDIA)
+"""
+import os
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM, MANDATORY
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.modules import get_software_root
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_XALT(ConfigureMake):
+    """Support for building and installing XALT."""
+
+    @staticmethod
+    def extra_options():
+        extra_vars = {
+            'config_py': [None, "XALT site filter file", MANDATORY],
+            'executable_tracking': [True, "Enable executable tracking", CUSTOM],
+            'gpu_tracking': [None, "Enable GPU tracking", CUSTOM],
+            'logging_url': [None, "Logging URL for transmission", CUSTOM],
+            'mysql': [False, "Build with MySQL support", CUSTOM],
+            'scalar_sampling': [True, "Enable scalar sampling", CUSTOM],
+            'syshost': [None, "System name", MANDATORY],
+            'transmission': [None, "Data tranmission method", MANDATORY],
+        }
+        return ConfigureMake.extra_options(extra_vars)
+
+    def configure_step(self):
+        """Custom configuration step for XALT."""
+
+        # By default, XALT automatically appends 'xalt/<version>' to the
+        # prefix, i.e., --prefix=/opt will actually install in
+        # /opt/xalt/<version>.  To precisely control the install prefix and
+        # not append anything to the prefix, use the configure option
+        # '--with-siteControlledPrefix=yes'.
+        # See https://xalt.readthedocs.io/en/latest/050_install_and_test.html
+        self.cfg.update('configopts', '--with-siteControlledPrefix=yes')
```

Comment:
You can avoid the use of this configuration option, you do something similar to what [we do](https://github.com/eth-cscs/production/blob/da8e1f42dcfe66511778750101d8770b08513962/easybuild/easyblocks/xalt.py#L46)[